cdap-tms/src/main/java/co/cask/cdap/messaging/client/ClientMessagingService.java

```diff
@@ -0,0 +1,511 @@
+    TopicId topicId = topicMetadata.getTopicId();
+
+    HttpRequest request = HttpRequest
+      .put(createURL(createTopicPath(topicId) + "/properties"))
+      .withBody(GSON.toJson(topicMetadata.getProperties()))
+      .build();
+    HttpResponse response = HttpRequests.execute(request, HTTP_REQUEST_CONFIG);
+
+    if (response.getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+      throw new TopicNotFoundException(topicId);
+    }
+    handleError(response, "Failed to update topic " + topicId);
+  }
+
+  @Override
+  public void deleteTopic(TopicId topicId) throws TopicNotFoundException, IOException {
+    HttpRequest request = HttpRequest
+      .delete(createURL(createTopicPath(topicId)))
+      .build();
+    HttpResponse response = HttpRequests.execute(request, HTTP_REQUEST_CONFIG);
+
+    if (response.getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+      throw new TopicNotFoundException(topicId);
+    }
+    handleError(response, "Failed to update topic " + topicId);
+  }
+
+  @Override
+  public TopicMetadata getTopic(TopicId topicId) throws TopicNotFoundException, IOException {
+    HttpRequest request = HttpRequest
+      .get(createURL(createTopicPath(topicId)))
+      .build();
+    HttpResponse response = HttpRequests.execute(request, HTTP_REQUEST_CONFIG);
+
+    if (response.getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+      throw new TopicNotFoundException(topicId);
+    }
+    handleError(response, "Failed to update topic " + topicId);
+
+    Map<String, String> properties = GSON.fromJson(response.getResponseBodyAsString(), TOPIC_PROPERTY_TYPE);
+    return new TopicMetadata(topicId, properties);
+  }
+
+  @Override
+  public List<TopicId> listTopics(NamespaceId namespaceId) throws IOException {
+    HttpRequest request = HttpRequest
+      .get(createURL(namespaceId.getNamespace() + "/topics"))
+      .build();
+    HttpResponse response = HttpRequests.execute(request, HTTP_REQUEST_CONFIG);
+
+    handleError(response, "Failed to list topics in namespace " + namespaceId);
+    List<String> topics = GSON.fromJson(response.getResponseBodyAsString(), TOPIC_LIST_TYPE);
+    List<TopicId> result = new ArrayList<>(topics.size());
+    for (String topic : topics) {
+      result.add(namespaceId.topic(topic));
+    }
+
+    return Collections.unmodifiableList(result);
+  }
+
+  @Override
+  public MessageFetcher prepareFetch(TopicId topicId) throws TopicNotFoundException, IOException {
+    return new ClientMessageFetcher(topicId);
+  }
+
+  @Nullable
+  @Override
+  public RollbackDetail publish(StoreRequest request) throws TopicNotFoundException, IOException {
+    HttpResponse response = performWriteRequest(request, true);
+
+    byte[] body = response.getResponseBody();
+    if (body.length == 0) {
+      return null;
+    }
+
+    // It has rollback detail, verify the content-type and decode it
+    verifyContentType(response.getHeaders().asMap(), "avro/binary");
```

Comment:
I thought about it, however the string is used in both client and server side. To do it right, have to define a ContentTypes class for that. I found it a bit overkill for now.